packages/@aws-cdk/aws-appsync/test/appsync-grant.test.ts

```diff
@@ -0,0 +1,419 @@
+                { Ref: 'AWS::Partition' },
+                ':appsync:',
+                { Ref: 'AWS::Region' },
+                ':',
+                { Ref: 'AWS::AccountId' },
+                ':apis/',
+                { 'Fn::GetAtt': [ 'API62EA1CFF', 'ApiId' ] },
+                '/types/Subscription/fields/subscribe',
+              ]],
+            },
+          },
+        ],
+      },
+    });
+  });
+});
+
+describe('grantType Permissions', () => {
+
+  test('grantType provides [type param]/* permissions when called without `fields` argument', () => {
+    // WHEN
+    api.grantType(role, appsync.IamResource.ofType('customType'));
+
+    // THEN
+    expect(stack).toHaveResourceLike('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: 'appsync:GraphQL',
+            Resource: {
+              'Fn::Join': [ '', [
+                'arn:',
+                { Ref: 'AWS::Partition' },
+                ':appsync:',
+                { Ref: 'AWS::Region' },
+                ':',
+                { Ref: 'AWS::AccountId' },
+                ':apis/',
+                { 'Fn::GetAtt': [ 'API62EA1CFF', 'ApiId' ] },
+                '/types/customType/*',
+              ]],
+            },
+          },
+        ],
+      },
+    });
+  });
+
+  test('grantType provides fields/[field param] permissions when called with `field` argument', () => {
+    // WHEN
+    api.grantType(role, appsync.IamResource.ofType('customType', ['attribute']));
+
+    // THEN
+    expect(stack).toHaveResourceLike('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: 'appsync:GraphQL',
+            Resource: {
+              'Fn::Join': [ '', [
+                'arn:',
+                { Ref: 'AWS::Partition' },
+                ':appsync:',
+                { Ref: 'AWS::Region' },
+                ':',
+                { Ref: 'AWS::AccountId' },
+                ':apis/',
+                { 'Fn::GetAtt': [ 'API62EA1CFF', 'ApiId' ] },
+                '/types/customType/fields/attribute',
+              ]],
+            },
+          },
+        ],
+      },
+    });
+  });
+});
+
+describe('grantFullAccess Permissions', () => {
+
+  test('grantFullAccess provides /* permissions', () => {
+    // WHEN
+    api.grantFullAccess(role);
+
+    // THEN
+    expect(stack).toHaveResourceLike('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: 'appsync:GraphQL',
+            Resource: {
+              'Fn::Join': [ '', [
+                'arn:',
+                { Ref: 'AWS::Partition' },
+                ':appsync:',
+                { Ref: 'AWS::Region' },
+                ':',
+                { Ref: 'AWS::AccountId' },
+                ':apis/',
+                { 'Fn::GetAtt': [ 'API62EA1CFF', 'ApiId' ] },
+                '/*',
+              ]],
+            },
+          },
+        ],
+      },
+    });
+  });
+});
```

Comment:
I _love_ those tests now!

However, I think we're missing a test for multiple fields?